build/docker/Dockerfile-develop

```diff
@@ -0,0 +1,52 @@
+# This dockerfile allows you to build and run OWASP ZAP using the latest source code
+# To build use a command line:
+#	docker build -t owasp/zap2docker-dev -f Dockerfile-develop .
+# To run use a command like
+# 	docker run -it owasp/zap2docker-dev bash
+#
+# /zap/src	will contain the ZAP repos with the latest source code (from when the container was built)
+# /zap/zap will contain a version of ZAP built from the source code
+#
+# If you change any of the ZAP extensions then you can rebuild them using:
+#   cd /zap/src/zaproxy && pwd && ls && ant -f build/build.xml deploy-weekly-addons
+# Note that this will just build and deploy the add-ons included in the weekly release,
+# see the relevant zap-extension build files for building and deploying other add-ons.
+#
+# If you change the ZAP core you can rebuild it using:
+#   cd /zap/src/zaproxy && pwd && ls && ant -f build/build.xml day-stamped-release
+# The /zap/zap directory is a symbolic link so will now link to the version you have just built.
+
+FROM openjdk:8-jdk
+MAINTAINER Simon Bennetts "psiinon@gmail.com"
+
+# Install ant for building and vim so that its easy to make quick changes 
+RUN apt-get update && apt-get install -y ant && apt-get install -y vim && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN useradd -d /home/zap -m -s /bin/bash zap 
+RUN echo zap:zap | chpasswd
+RUN mkdir -p /zap/src 
+RUN chown -R zap /zap && \
+	chgrp -R zap /zap
+WORKDIR /zap/src
+
+USER zap
+
+# Pull the ZAP repos
+RUN git clone https://github.com/zaproxy/zaproxy.git && \
+	git clone https://github.com/zaproxy/zap-extensions.git && \
+	git clone --branch beta https://github.com/zaproxy/zap-extensions.git zap-extensions_beta && \
+	git clone --branch alpha https://github.com/zaproxy/zap-extensions.git zap-extensions_alpha && \
+	git clone https://github.com/zaproxy/zap-core-help.git
+
+# Build ZAP
+RUN cd /zap/src/zaproxy && pwd && ls && \
+	ant -f build/build.xml deploy-weekly-addons && \
+	ant -f build/build.xml day-stamped-release
+	
+RUN chmod +x /zap/src/zaproxy/build/zap/zap.sh
+	
+RUN	ln -s /zap/src/zaproxy/build/zap /zap
+
+# Sanity check that we can run ZAP	
+RUN cd /zap/zap && ./zap.sh -version
```

Comment:
The idea of this image was to allow people to start hacking on the ZAP source code easily, so deleting the source wouldnt be good ;)
However ... we could just have another image which was based on this and just deleted the source. And, thinking about it more, this should include all of the other sw that the weekly release includes (if its to replace the weekly release).
So the image should really be much closer to the weekly one :/
I'll do some more playing ...